framework/Source/Sparkles.swift

```diff
@@ -28,6 +28,20 @@ public class Sparkles: OperationGroup {
         didSet { perlinNoiseEffect.time = time }
     }
     // LightExtractor
+    public var faceMaskImage: UIImage? {
+        didSet {
+            guard faceMaskImage != oldValue else { return }
+            faceMaskInput = faceMaskImage.map { PictureInput(image: $0) }
+        }
+    }
+    private var faceMaskInput: PictureInput? {
+        willSet { faceMaskInput?.removeAllTargets() }
+        didSet {
+            guard let faceMaskInput else { return }
+            handleFaceMaskUpdated(faceMaskInput)
+        }
+    }
```

Comment:
question: could this run without a mask?

We might need to handle nil case, just give a empty mask

Reply:
Sure that will be better 👍
I do this in PicCollage, but just like your suggestion, it makes more sense to create an empty Input inside this filter.
I'll try this later on; thanks!

Reply:
I'll update this in the PR for the face mask

Reply:
Hi @yyjim, I've update the code at here: [fix: set up empty pictureInput when the faceMaskImage is nil](https://github.com/cardinalblue/GPUImage3/pull/19/commits/562f8bd3dc0d85ef328081b1682661ed1803a6c1)